frontend/vre/utils/jsonld.model.js

```diff
@@ -1,11 +1,102 @@
 import {APICollection} from "./api.model";
 
 /**
- * Generic subclass of APICollection that parses incoming JSON-LD to a
- * list of subjects.
+ * The @graph property inside JSON-LD
+ * @typedef {Object} JSONLDGraph
+ */
+
+/**
+ * An individual subject definition inside JSON-LD
+ * @typedef {Object} JSONLDSubject
+ */
+
+export var JsonLdModel = Backbone.Model.extend({
+    idAttribute: '@id',
+
+})
+
+/**
+ * Generic subclass of APICollection that parses incoming compacted JSON-LD to an
+ * array of all subjects. The contents of subjects are left unchanged.
  */
 export var JsonLdCollection = APICollection.extend({
+    model: JsonLdModel,
     parse: function(response) {
+        if (!response.hasOwnProperty("@graph")) {
+            throw "Response has no @graph key, is this JSON-LD in compacted form?";
+        }
         return response["@graph"];
     }
 });
+
+/**
+ * Return a nested version of a given subject by adding to it the objects
+ * it refers to if they are found in the graph.
+ * The subject passed to this function as an argument is not changed.
+ * @param subjectsByID{Dictionary<JSONLDSubject>} - The full contents of the graph in JSON-LD
+ * @param subject{JSONLDSubject} - The subject including its predicates and objects to create a nested version of
+ * @param parentSubjectIDs{Array<String>} - For internal use of recursive function; leave undefined
+ * @returns {Object}
+ */
+export function nestSubject(subjectsByID, subject, parentSubjectIDs=undefined) {
+    if (typeof parentSubjectIDs === "undefined") {
+        parentSubjectIDs = [subject["@id"]];
+    } else {
+        parentSubjectIDs = Array.from(parentSubjectIDs);
+        parentSubjectIDs.push(subject["@id"]);
+    }
+    const transformedSubject = _.clone(subject);
+    for (let property of Object.keys(subject)) {
+        if (subject[property].hasOwnProperty("@id")) {
+            // This is a reference to another subject
+            const refereedSubject = subjectsByID[subject[property]["@id"]];
+            if (refereedSubject && !(parentSubjectIDs.includes(refereedSubject["@id"]))) {
+                /* If the refereed subject was found in the graph, use it as replacement.
+                   Only do this if the subject is not the same as the one we started with,
+                   to avoid an endless loop. (Alternative would be to create a circular reference) */
```

Comment:
```suggestion
                   Only do this if we have not visited the same subject before,
                   to avoid an endless loop. (Alternative would be to create a circular reference) */
```